Public/Src/Cache/ContentStore/Distributed/NuCache/RocksDbContentLocationDatabase.cs

```diff
@@ -588,27 +602,28 @@ public override GetContentHashListResult GetContentHashList(OperationContext con
                     {
                         if (store.TryGetValue(key, out var data, nameof(Columns.Metadata)))
                         {
-                            var current = DeserializeContentHashList(data);
-
-                            if (!current.Equals(expected))
+                            var current = DeserializeMetadataEntry(data);
+                            if (!current.ContentHashListWithDeterminism.Equals(expected))
                             {
                                 return false;
                             }
                         }
-
-                        store.Put(key, SerializeContentHashList(replacement), nameof(Columns.Metadata));
+                        
+                        var replacementMetadata = new MetadataEntry(replacement, Clock.UtcNow.ToFileTimeUtc());
+                        store.Put(key, SerializeMetadataEntry(replacementMetadata), nameof(Columns.Metadata));
                     }
 
                     return true;
                 });
         }
         
         /// <inheritdoc />
-        public override AddOrGetContentHashListResult AddOrGetContentHashList(OperationContext context, StrongFingerprint strongFingerprint, ContentHashListWithDeterminism contentHashListWithDeterminism)
+        public override Task<AddOrGetContentHashListResult> AddOrGetContentHashListAsync(OperationContext context, StrongFingerprint strongFingerprint, ContentHashListWithDeterminism contentHashListWithDeterminism, IContentSession contentSession,

```

Comment:
>contentSession [](start = 231, length = 14)

I think we should add a callback to check this thing, instead of referencing contentsession.

Reply:
I moved this logic into the RocksDbMemoizationStore, it is better suited to that part (dealing with determinism and all that)

---
In reply to: [305161776](https://github.com/microsoft/BuildXL/pull/621#discussion_r305161776) [](ancestors = 305161776)